### PR TITLE
adding screenshot method to the simulator

### PR DESCRIPTION
**Pull request by nturgut**
adding screenshot method to the simulator. It will be used instead of the method here: https://github.com/flutter/engine/pull/20963/files#diff-2a1bd9f484e3b58f6dcc82805bfd9d06R186

related: https://github.com/flutter/flutter/issues/65670

### CODE REVIEW

File: packages/simulators/lib/simulator_manager.dart [hunk 1]
```diff
@@ -228,6 +228,28 @@ class IosSimulator {
...
+
+    if (versionResult.exitCode != 0) {
+      io.stdout
+          .writeln('Info: Failed to run xcrun screenshot on iOS simulator. '
+              'Simulator id: $id');
+      io.stderr
+          .writeln('ERROR: Failed to run xcrun screenshot on iOS simulator. '
+              'Simulator id: $id');
+      throw Exception('Failed to run xcrun screenshot on iOS simulator for '
+          'simulator id: $id');
```
Comment (yjbanov) [inline packages/simulators/lib/simulator_manager.dart:249]: I think this is way too many notifications. Why is throwing an exception not sufficient?

File: packages/simulators/lib/simulator_manager.dart [hunk 2]
```diff
@@ -228,6 +228,28 @@ class IosSimulator {
     return;
   }
 
+  /// Take the screenshot of the simulator.
+  ///
+  /// Note that, this method takes screenshot of the entire simulator, not the
```
Comment (yjbanov) [inline packages/simulators/lib/simulator_manager.dart:233]: nit: "note that" isn't necessary.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

File: packages/simulators/lib/simulator_manager.dart (hunk 1 above)
Comment (nturgut) [reply packages/simulators/lib/simulator_manager.dart:243]: done.

I was asked on the other PR: https://github.com/flutter/engine/pull/20963#discussion_r486648641

File: packages/simulators/lib/simulator_manager.dart (hunk 2 above)
Comment (nturgut) [reply packages/simulators/lib/simulator_manager.dart]: done. thanks for sharing the link.

File: packages/simulators/lib/simulator_manager.dart (hunk 1 above)
Comment (yjbanov) [reply packages/simulators/lib/simulator_manager.dart:243]: Here I was referring to printing the same message 3 times. In the other PR I was referring to printing the stdout/stderr of the child process. We could do that here as well so we have more debug info when xcrun fails.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**